src/Model/List/PaginatedList.php

```diff
@@ -557,4 +549,15 @@ public function getRequest()
     {
         return $this->request;
     }
+
+    private function getListForCurrentPage()
+    {
+        $pageLength = $this->getPageLength();
+        if ($this->limitItems && $pageLength) {
+            $tmptList = clone $this->list;
+            return$tmptList->limit($pageLength, $this->getPageStart());
```

Comment:
```suggestion
            return $tmptList->limit($pageLength, $this->getPageStart());
```
Was this code not called at all during unit tests? Or is it somehow valid?

Reply:
This is, surprisingly, perfectly valid PHP. But yeah it's bad lol. Fixed.
I'm a bit sad our linters didn't pick up on it.